### PR TITLE
Move multi-step training into TrainingConfig with per-step IS correction

### PR DESCRIPTION
**Pull request by kfallah**
## Summary
- move multi-step training controls (`steps_per_batch`, `feedback_repetitions`) from eval-owned settings into `TrainingConfig`
- remove eval-side sub-step loop and pass typed `training` config through `FeedbackItem` in each `/v1/feedback` request
- execute multi-step updates inside training engines (local/modal + tinker)
- recompute behavior-policy logprobs after each optimizer step for off-policy importance reweighting
- include engine metadata (`steps_per_batch_applied`, per-step metrics) and wire eval `sub_step_count` to that metadata
- update eval Hydra schema/config/docs and related tests

## Key Implementation Notes
- added strict `TrainingConfig` fields:
  - `steps_per_batch`
  - `feedback_repetitions`
- introduced Hydra-safe `EvalTrainingConfig` and convert to runtime `TrainingConfig` in `build_harness_config`
- tinker engine now refreshes student logprobs between steps using `save_weights_and_get_sampling_client_async`

## Validation
- `uv run ruff check claas/ tests/ --fix`
- `uv run pytest tests/ -q -m "not integration"`
  - result: `109 passed, 26 skipped, 5 deselected`
- `uv run ty check`
  - unresolved-import diagnostics for heavy runtime deps (`torch`, `tinker`, `transformers`) are expected in this environment

## Release Notes

* **New Features**
  * Added support for multi-step training per batch with configurable `steps_per_batch` parameter
  * Added `feedback_repetitions` configuration option for enhanced training control
  * New metric `steps_per_batch_applied` tracks actual steps executed per batch

* **Documentation**
  * Updated configuration structure to use nested training block for training-specific parameters

* **Refactor**
  * Reorganized configuration hierarchy to consolidate training settings under dedicated training section

### CODE REVIEW

**Comment by kfallah** [comment]
@codex review

File: claas/eval/config.py [hunk 1]
```diff
@@ -21,6 +23,8 @@
 def build_harness_config(eval_cfg: EvalConfig) -> HarnessConfig:
     """Post-process EvalConfig → HarnessConfig (no secrets)."""
     fields = dataclasses.asdict(eval_cfg)
+    training_fields = fields.pop("training")
+    fields["training"] = TrainingConfig(**training_fields)
```
[the remark this answers is not in the document]
Comment (kfallah) [reply claas/eval/config.py]: @codex `TrainingConfig` is a `@dataclass`, not a Pydantic model — `dataclasses.asdict()` recursively converts nested dataclasses by design. This doesn't break JSON serialization.

*Comment generated by Claude Code*

File: claas/eval/runner.py [hunk 2]
```diff
@@ -81,13 +81,15 @@ async def _submit_feedback(
...
+            steps_per_batch_applied=metadata["steps_per_batch_applied"],
         )
 
     return LocalDistillMetrics(
         distill_loss=metadata.get("distill_loss"),
         kl_reg=metadata.get("kl_reg"),
         mean_is_ratio=metadata.get("mean_is_ratio"),
         clip_fraction=metadata.get("clip_fraction"),
+        steps_per_batch_applied=metadata["steps_per_batch_applied"],
     )
```
[the remark this answers is not in the document]
Comment (kfallah) [reply claas/eval/runner.py]: Fixed in [`26fa419`](https://github.com/kfallah/CLaaS/commit/26fa419d11c3b7a2be479593c619710a63ba1e19)

*Comment generated by Claude Code*

File: claas/core/types.py [hunk 3]
```diff
@@ -32,6 +32,8 @@ class TrainingConfig:
     max_grad_norm: float = 1.0
     kl_reg_weight: float = 0.0
     teacher_top_k: int = 100
+    steps_per_batch: int = 4
```
[the remark this answers is not in the document]
Comment (kfallah) [reply claas/core/types.py:35]: Fixed in [`c7eb678`](https://github.com/kfallah/CLaaS/commit/c7eb678dcfb0ff52fc9762ac643bc0ee83d276f5)

*Comment generated by Claude Code*

**Comment by kfallah** [comment]
@codex review
